storage/src/vespa/storage/storageserver/bouncer.cpp

```diff
@@ -283,25 +284,51 @@ Bouncer::onDown(const std::shared_ptr<api::StorageMessage>& msg)
     return false;
 }
 
+namespace {
+
+lib::NodeState
+deriveNodeState(const lib::NodeState &reportedNodeState,
+                const lib::NodeState &currentNodeState,
+                bool useWantedStateIfPossible)
+{
+    if (useWantedStateIfPossible) {
+        // If current node state is more strict than our own reported state,
+        // set node state to our current state
+        if (reportedNodeState.getState().maySetWantedStateForThisNodeState(currentNodeState.getState())) {
+            return currentNodeState;
+        }
+    }
+    return reportedNodeState;
+}
+
+}
+
 void
 Bouncer::handleNewState()
 {
     vespalib::LockGuard lock(_lock);
-    _nodeState = *_component.getStateUpdater().getReportedNodeState();
+    const auto reportedNodeState = *_component.getStateUpdater().getReportedNodeState();
     const auto clusterStateBundle = _component.getStateUpdater().getClusterStateBundle();
     const auto &clusterState = *clusterStateBundle->getBaselineClusterState();
     _clusterState = &clusterState.getClusterState();
-    if (_config->useWantedStateIfPossible) {
-        // If current node state is more strict than our own reported state,
-        // set node state to our current state
-        lib::NodeState currState = clusterState.
-                getNodeState(lib::Node(_component.getNodeType(),
-                                       _component.getIndex()));
-        if (_nodeState.getState().maySetWantedStateForThisNodeState(
-                    currState.getState()))
-        {
-            _nodeState = currState;
-        }
+    const lib::Node node(_component.getNodeType(), _component.getIndex());
+    const bool useWantedStateIfPossible = _config->useWantedStateIfPossible;
```

Comment:
Consider removing config option and always assume this is true. As far as I know, no one has ever wanted or needed to use anything but the default here.